dalek/cleaners/storage_account_lock.go

```diff
@@ -0,0 +1,62 @@
+package cleaners
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2020-05-01/managementlocks"
+	"github.com/jackofallops/azurerm-dalek/clients"
+	"github.com/jackofallops/azurerm-dalek/dalek/options"
+)
+
+var _ ResourceGroupCleaner = removeLocksFromStorageAccountCleaner{}
+
+type removeLocksFromStorageAccountCleaner struct {
+}
+
+func (removeLocksFromStorageAccountCleaner) Name() string {
+	return "Storage Account Locks"
+}
+
+func (removeLocksFromStorageAccountCleaner) Cleanup(ctx context.Context, id commonids.ResourceGroupId, client *clients.AzureClient, opts options.Options) error {
+	storageAccounts, err := client.ResourceManager.StorageAccountsClient.ListByResourceGroupComplete(ctx, id)
+	if err != nil {
+		log.Printf("[DEBUG] Error listing storage accounts in resource group %q: %+v", id.ResourceGroupName, err)
+		return err
```

Comment:
To prevent the dalek from exiting early, we tend to not return errors

Reply:
Maybe you're thinking about the subscription cleaners? (this is a resource group cleaner)
```
if errors := client.ResourceManager(ctx); len(errors) != 0 {
errList := make([]string, 0)
for _, e := range errors {
	errList = append(errList, e.Error())
}

return fmt.Errorf("processing Resource Manager: %+v", strings.Join(errList, "\n"))
}
```